src/operations/blas1_trees.hpp

```diff
@@ -374,6 +465,25 @@ SYCL_BLAS_INLINE typename BinaryOp<operator_t, lhs_t, rhs_t>::value_t
 BinaryOp<operator_t, lhs_t, rhs_t>::eval(cl::sycl::nd_item<1> ndItem) {
   return BinaryOp<operator_t, lhs_t, rhs_t>::eval(ndItem.get_global_id(0));
 }
+
+template <typename operator_t, typename lhs_t, typename rhs_t>
+template <int vector_size>
+SYCL_BLAS_INLINE cl::sycl::vec<
+    typename BinaryOp<operator_t, lhs_t, rhs_t>::value_t, vector_size>
+BinaryOp<operator_t, lhs_t, rhs_t>::eval_vector(
+    typename BinaryOp<operator_t, lhs_t, rhs_t>::index_t i) {
+  return operator_t::eval(lhs_.template eval_vector<vector_size>(i),
```

Comment:
Should it be eval_vector ?

Reply:
@aacostadiaz From what I saw the [Operators](https://github.com/codeplaysoftware/sycl-blas/blob/master/src/operations/blas_operators.hpp#L225) don't seem to need a specialization for the `eval` functions as they're already templated by the `lhs` and `rhs` types and in the case of vector inputs they'll handle them  (as the `sycl::vec` interface already overloads the main operators used here). What do you think?